### DISCUSSION TITLE
Unique constraint failed on the constraint for one to many Id field

### DISCUSSION THREAD

**Discussion by anna-heylibby** [prisma/prisma#22491] [Q&A] [answered]
### Question

I have a one-to-many relationship defined between two Prisma models: Agent and Contact, where one Agent should have many Contacts.  However, the database shows that the AgentId on the Contact has been created as a Unique Key and when I try to create multiple Contacts with the same Agent Id, I get the error: 
return prisma.contact.create(
Unique constraint failed on the constraint: `Contact_agentId_key`

How do I remove the uniqueness constraing on the agentId? 

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
model Agent {
  id            String  @id @default(uuid())
  org           Org?    @relation(fields: [orgId], references: [id])
  orgId         String?
  contacts      Contact[]
  businessName  String
}

model Contact {
  id              String    @id @default(uuid())
  agent           Agent     @relation(fields: [agentId], references: [id]) 
  agentId         String 
  name            String
}
```

**Comment by ludralph** [comment]
Hi @anna-heylibby 👋 

Thank you for raising this question.

Can you please share the code you use to create multiple Contacts with the same Agent Id?

**Reply by anna-heylibby** [reply]
@ludralph, thanks for the response!

Here is the contactClient: 

```
export const createContact = async (
  prisma: PrismaClient,
  id: string,
  agentId: string,
  name: string,
) => {
  return prisma.contact.create({
    data: {
      id,
      agentId,
      name,
    },
  })
}
```

And here is the code snippet in the lambda function that calls the client. 

```
async function postContact(prisma: PrismaClient, event: APIGatewayProxyEvent) {
    const body = JSON.parse(event.body)
    const uid = body.id
    const newContact = await createContact(
        prisma,
        uid,
        body.agentId,
        body.name,
     )       
     return newContact
  }
```

**Comment by anna-heylibby** [comment, marked as answer]
I was able to fix this, but not in an entirely satisfactory way. Only thing that did it was to remove the relationship entirely, do a prisma migrate and then add it back exactly the same way we had it before.
